src/util/util.ts

```diff
@@ -37,3 +37,9 @@ export const isSubstringOf = (substring: string, source: string): boolean =>
   pipe(source, S.toLowerCase, S.includes(S.toLowerCase(substring)))
 
 export const isError = (value: unknown): value is Error => value instanceof Error
+
+export const escapeRegExp = (str?: string): string => {
+  const regExp = /[\\^"'$.*+?()[\]{}|]/g
```

Comment:
Did you test it on a SPARQL query ? Escaping `$`, `.` and so on will not work.

> FILTER ( contains(lcase(str(?title)), "\." ) )
>  Encountered: "." (46), after : "\"\\"

Cf. https://www.w3.org/TR/sparql11-query/#grammarEscapes

Reply:
@ccamel thanks for the specification link.  We probably only need to manage the escape sequence provided by the Sparql spec.
The function would be:
```ts
export const escapeSparqlStr = (str?: string): string => {
  const sparqlEscapeRegExp = /[\t\n\r\b\f"\\']/g
  const escapeCharsMap: { [key: string]: string } = {
    '\t': '\\t',
    '\n': '\\n',
    '\r': '\\r',
    '\b': '\\b',
    '\f': '\\f',
    '"': '\\"',
    `'`: `\\'`,
    '\\': '\\\\',
  }

  return str ? str.replace(sparqlEscapeRegExp, (char) => escapeCharsMap[char]) : '';
}
```
It would be easier to read if we use a Map or something similar because we don't have that many characters to escape and the replacement string doesn't follow the same pattern. So using the entire match (`$&`) with a leading slash doesn't cover all cases.
I used backticks in the exemple for escaping single quote for readability.